streams/src/main/java/org/apache/kafka/streams/state/internals/WindowStoreSupplier.java

```diff
@@ -0,0 +1,21 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.state.internals;
+
+public interface WindowStoreSupplier {
```

Comment:
I'm pretty sure adding a `PersistentWindowFactory` would be a breaking change. `Stores` is a public API and to make this work we'd need to change `PersistentKeyValueFactory.windowed` to return a `PersistentWindowFactory` - this could break existing usages. Would this need another KIP?